detox/src/devices/android/ADB.js

```diff
@@ -13,8 +13,22 @@ class ADB {
   }
 
   async devices() {
-    const output = (await this.adbCmd('', 'devices', { verbosity: 'high' })).stdout;
-    return this.parseAdbDevicesConsoleOutput(output);
+    const parser = await this._adbDevices();
+    const devices = [];
+    for (let device = await parser.parseNext(); device; device = await parser.parseNext()) {
```

Comment:
Not sure, but isn't there something wrong with:
```
for (let device = await parser.parseNext(); device; device = await parser.parseNext())
```
?

You have two `parseNext()` methods in one row, so you will skip every 2nd device, no? read device, do something, read device, read device, do something -> 1, 3, 5, 7, ... ?